### PR TITLE
[POC] add a way to change the xpath prefix

### LINKED ISSUE #649: WebAssert (and getPage) should be able to dive into xml documents too
As (not really) discussed in #408, it should be possible to query something else than `//html`-prefixed xpath queries.

I have a working diff here and wanted to know if you'd be interested in looking at it.

Cheers!

### PR DESCRIPTION
**Pull request by docteurklein**
Fixes #649.
needs tests. Thoughts @stof ?

Usage:

       $mink->assertSession(null, '//opendata')->elementExists('xpath', $selector);

My first impression is that I don't like to mess with optional `$xpath` arguments :|
Also, I changed the signature of some methods.

### CODE REVIEW

**Comment by aik099** [comment]
Maybe we should approach this by specifying custom DocumentElement instance instead of passing `//html` all over the place?

Currently there is single `DocumentElement` instance per session and you can't change it over time to query XML and then HTML documents.

**Comment by docteurklein** [comment]
I thought about a `setPage` on the session object, but it doesn't sounds right.
But passing `//html` all over the place doesn't sound better :)

One could then do: 

    $session->setPage(new DocumentElement('//root', $session));

and basta.

File: src/Element/DocumentElement.php [hunk 1]
```diff
@@ -17,6 +19,21 @@
...
 {
+    private $xpath;
+
+    /**
+     * Initializes node element.
+     *
+     * @param string  $xpath   element xpath
+     * @param Session $session session instance
+     */
+    public function __construct($xpath, Session $session)
```
Comment (stof) [inline src/Element/DocumentElement.php]: this signature change is a BC break.

File: src/Session.php [hunk 2]
```diff
@@ -118,9 +117,9 @@ public function getDriver()
      *
      * @return DocumentElement
      */
-    public function getPage()
+    public function getPage($xpath = '//html')
     {
-        return $this->page;
+        return $this->page = $this->page ?: new DocumentElement($xpath, $this);
```
Comment (stof) [inline src/Session.php]: this logic is broken in case the page already exists and you pass a different XPath

**Comment by stof** [comment]
I would vote against this as it seems out of the scope of Mink and creates a bunch of issues

File: src/Element/DocumentElement.php (hunk 1 above)
Comment (stof) [inline src/Element/DocumentElement.php]: see the testsuite btw

**Comment by docteurklein** [comment]
@aik099 @stof seems legit. Do you have an idea how I could achieve the same goal without this?

**Comment by aik099** [comment]
Closing, because associated issue was closed. See associated issue discussion for an answer.

**Pull request closed without merging**